set_up_environment/first_steps.rst

```diff
@@ -0,0 +1,209 @@
+
+A good way to start is to clone the `Aplus course Template`_ with
+git and begin to work on it. In Aalto, you could use
+`version.aalto.fi repo <https://version.aalto.fi/gitlab/course/aplus-manual>`_. Please note that if you don't have
+access, then you can request access to the *Aalto IT services*. In that case, you probably should be part of
+apluslms-cs@aalto.fi email list too.
+
+To clone, execute the following command:
+
+.. code-block:: sh
+
+    git clone git@version.aalto.fi:course/aplus-manual.git
+    cd aplus-manual
+
+In public, you could use `GitHub repo <https://github.com/apluslms/course-templates/>`_.
+
+.. code-block:: sh
+
+    git clone https://github.com/apluslms/course-templates.git
+    cd course-templates
+
+Next step, you need to get :code:`a-plus-rst-tools`.
+
+.. code-block:: sh
+
+    git submodule init
+    git submodule update
+
+Now all the course material is in RST format. Every time the RST part changes,
+you need to recompile it. You must do it also the first time. Open your terminal, go to the course directory and type
+the following command.
+
+.. code-block:: sh
+
+    ./docker-compile.sh
+
+This command runs Sphinx inside a Docker container. Sphinx reads all RST files.
+It produces compilation results into new directory ``_build``. That contains
+three subdirectories. ``doctrees`` has intermediate results of compilation.
+``html`` has the HTML versions of the RST documentation. ``yaml`` contains
+configuration files in `YAML format <https://en.wikipedia.org/wiki/YAML>`_
+and they are also meant for A+.
+
+For now, you can test the result by running A+ on your own computer:
+
+.. code-block:: sh
+
+    ./docker-up.sh
+
+Now A+ runs at http://localhost:8000. Open that address with your web browser.
+You must login by the maintenance login showing on the front page. The available
+users are `root`:`root` and `student`:`student`.  The default course is created
+from the material.
+
+Workflow
+--------
+
+The usual workflow for creating/editing a course is the following:
+
+1. Open your text editor, a terminal and a web browser.
+2. Edit some RST files in your text editor.
+3. Give command ``./docker-compile.sh`` in the terminal.
+4. Give command ``./docker-up.sh`` in the terminal.
+5. Go to ``http://localhost:8000/`` in the web browser to view A+
+    running on your machine.
+6. Examine the changes you made in A+.
+7. Press :kbd:`Q` or :kbd:`Ctrl` ``+`` :kbd:`C` in the terminal to quit and remove data
+    or :kbd:`S` or :kbd:`ESC` to quit and keep data.
+8. Go to step 2 if you wish to continue editing.
+9. Give command ``git add -u`` in the terminal to mark all changed files to be
+    added into your local git repository.
+10. Give command ``git commit -m "message"`` in terminal. Replace ``message`` with a short description (preferably less
+than 60 characters) on what you have done.
+11. Remember also to say ``git push`` a couple of times in a day. This copies your updates to the course material to the
```

Comment:
`git push` here assumes that the repo was cloned from a course repo instead of the aplus-manual repo. Previously, readers were instructed to clone the manual repo, but they are not supposed to push their own course contents there.